x-pack/plugins/security_solution/scripts/endpoint/resolver_generator_script.ts

```diff
@@ -298,8 +300,11 @@ async function main() {
 
     clientOptions = { node, tls: { ca: [ca] } };
   } else {
-    clientOptions = { node: argv.node };
+    clientOptions = { node };
```

Comment:
I don't understand why exactly this change is needed? It won't work if you provide the `n` argument before the `k` argument. Right? 

Looking purely at types, `ClientOptions` type has a `node` key that does accept a `string[]`. Also, looking at the script you're trying to run, are ES and Kibana running on the same port (`5601`)? 

The real question is, why are two different CLI arguments here (`n` for ES URL, `k` for Kibana URL) are getting combined into a single array as the `n` argument.

Reply:
Hey @ashokaditya , Sorry for my typos which created the confusion. I have updated the description after running script on main again. 


```
node scripts/endpoint/resolver_generator -ne 10 -nd 100 -ape 20 -k http://elastic_serverless:changeme@localhost:5601 -n http://elastic_serverless:changeme@localhost:9200
```
Above script  produces this array as the `node` value:

```
 info ES Url :  [
        "http://elastic:changeme@127.0.0.1:9200",
        "http://elastic:changeme@127.0.0.1:9200",
        "http://elastic_serverless:changeme@localhost:9200"
      ]
```
which in turn gives below error: 

```
{
  "options": {
    "redaction": {
      "type": "replace",
      "additionalKeys": []
    }
  },
  "name": "ResponseError",
  "meta": {
    "body": {
      "error": {
        "root_cause": [
          {
            "type": "security_exception",
            "reason": "unable to authenticate user [elastic] for REST request [/_data_stream/logs-endpoint.events.process-default]",
            "header": {
              "WWW-Authenticate": [
                "ApiKey",
                "Basic realm=\"security\" charset=\"UTF-8\""
              ]
            }
          }
        ],
        "type": "security_exception",
        "reason": "unable to authenticate user [elastic] for REST request [/_data_stream/logs-endpoint.events.process-default]",
        "header": {
          "WWW-Authenticate": [
            "ApiKey",
            "Basic realm=\"security\" charset=\"UTF-8\""
          ]
        }
      },
      "status": 401
    },
    "statusCode": 401,
    "headers": {
      "www-authenticate": [
        "ApiKey",
        "Basic realm=\"security\" charset=\"UTF-8\""
      ],
      "content-type": "application/json",
      "content-length": "507"
    },
    "meta": {
      "context": null,
      "request": {
        "params": {
          "method": "DELETE",
          "path": "_data_stream/logs-endpoint.events.process-default",
          "querystring": "",
          "headers": {
            "user-agent": "elasticsearch-js/8.13.1 (darwin 23.5.0-arm64; Node.js 20.13.1; Transport 8.4.1)",
            "x-elastic-client-meta": "es=8.13.1,js=20.13.1,t=8.4.1,hc=20.13.1",
            "authorization": "[redacted]",
            "accept": "application/vnd.elasticsearch+json; compatible-with=8,text/plain"
          }
        },
        "options": {},
        "id": 1
      },
      "name": "elasticsearch-js",
      "connection": {
        "url": "http://127.0.0.1:9200/",
        "tls": null,
        "id": "http://127.0.0.1:9200/",
        "timeout": 30000,
        "headers": {
          "user-agent": "elasticsearch-js/8.13.1 (darwin 23.5.0-arm64; Node.js 20.13.1; Transport 8.4.1)",
          "x-elastic-client-meta": "es=8.13.1,js=20.13.1,t=8.4.1,hc=20.13.1",
          "authorization": "[redacted]"
        },
        "deadCount": 0,
        "resurrectTimeout": 0,
        "_openRequests": 0,
        "weight": 333,
        "pool": {
          "_events": {},
          "_eventsCount": 0
        }
      },
      "attempts": 0,
      "aborted": false
    },
    "warnings": null
  }
}

```